lib/membrane_opus/parser.ex

```diff
@@ -70,15 +79,20 @@ defmodule Membrane.Opus.Parser do
   end
 
   @impl true
-  def handle_buffer(:input, %Buffer{payload: data}, ctx, state) do
+  def handle_buffer(:input, %Buffer{payload: data, pts: pts}, ctx, state) do
     {delimitation_processor, self_delimiting?} =
       Delimitation.get_processor(state.delimitation, state.input_delimitted?)
 
     case maybe_parse(
            state.buffer <> data,
-           state.pts,
+           if state.generate_best_effort_timestamps? do
+             state.pts
+           else
+             pts
+           end,
```

Comment:
AFAIK if `state.generate_best_effort_timestamps?` is set to false, but `state.buffer` is not empty, the first returned frame should have `pts` derived from the previous buffer, not from the current one, but you ignore `state.pts` anyway, while it may still be useful

(if `state.buffer` is not empty, it means, that first returned frame has been splitted between current payload and previous one),